src/transformers/models/blip/configuration_blip.py

```diff
@@ -0,0 +1,402 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Blip model configuration"""
+
+import copy
+import os
+from typing import Union
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+BLIP_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "Salesforce/blip-vqa-base": "https://huggingface.co/Salesforce/blip-vqa-base/resolve/main/config.json",
+    "Salesforce/blip-vqa-base-capfit": (
+        "https://huggingface.co/Salesforce/blip-vqa-base-capfit/resolve/main/config.json"
+    ),
+    "Salesforce/blip-image-captioning-base": (
+        "https://huggingface.co/Salesforce/blip-image-captioning-base/resolve/main/config.json"
+    ),
+    "Salesforce/blip-image-captioning-large": (
+        "https://huggingface.co/Salesforce/blip-image-captioning-large/resolve/main/config.json"
+    ),
+    "Salesforce/blip-retrieval-coco-base": (
+        "https://huggingface.co/Salesforce/blip-retrieval-coco-base/resolve/main/config.json"
+    ),
+    "Salesforce/blip-retrieval-coco-large": (
+        "https://huggingface.co/Salesforce/blip-retrieval-coco-large/resolve/main/config.json"
+    ),
+    "Salesforce/blip-retrieval-flikr-base": (
+        "https://huggingface.co/Salesforce/blip-retrieval-flikr-base/resolve/main/config.json"
+    ),
+    "Salesforce/blip-retrieval-flikr-large": (
+        "https://huggingface.co/Salesforce/blip-retrieval-flikr-large/resolve/main/config.json"
+    ),
+}
+
+
+class BlipTextConfig(PretrainedConfig):
```

Comment:
Copied from possible?

Reply:
Hmm not really since it has different / more argument than `CLIPTextConfig` or `BertConfig`